docs/2-fork-accountability.md

```diff
@@ -0,0 +1,163 @@
+On the other hand, if it does not consider p faulty, the monitor might not be able to find all the faulty processes that led to a fork.
+
+#### Asynchronous model
+
+This method would work but we want to make no assumptions regarding the communication between the monitor and the validators. 
+If we do not make any assumption regarding the communication, we would realize that it is impossible to design an accountability algorithm where the communication between monitor and validators is completely asynchronous.
+
+The reason is simple: in an asynchronous setting, the monitor cannot be sure a process P is indeed faulty if it did not receive P's message logs. 
+In this scenario, the accountability algorithm can only determine the faultiness of P by analyzing the messages logs sent by other processes and equivocation is the unique possible faultiness that can be detected for P.
+ 
+Equivocation is a faulty behavior that happens when a process sent more than one message with the same type in the same round but with different values.
+
+By receiving messages logs from correct processes, the monitor would be able to determine if some correct process received more than one message with the same type from another process p (i.e., if some process equivocated).  
+However, we have also shown that a process can be faulty in other ways and that would not be enough for designing an asynchronous accountability algorithm that catches all the faulty processes that led to a fork. 
+In fact, it can happen that, by not receiving message logs from one message, we might end up mistakenly detecting correct processes because we do not have enough information to determine if a process has enough justifications to have sent a Prevote/Precommit message (accuracy property violated). 
+
+If we want to design a correct accountability algorithm, we need to slightly modify the Tendermint consensus algorithm in order to correctly respect the accuracy property.
+As we said above, we need to catch the case where a fork happens without having any process equivocating.
+
+By looking at the other faultiness reasons, we can see that the critical case is when a process sends a Precommit message for a value v and then, in a later round, it sends a Prevote message for a different value than v.
+The asynchronous accountability algorithm would work correctly if processes can justify the sending of a Prevote message: a simple solution is to attach the justifications to the Prevote message itself so that the accountability algorithm can directly check if the message has been correctly sent.
+
+This would be enough because, intuitively, there will be at least one correct process that has received both messages and it would be possible to track down the origin and the justifications and find out if a process is really faulty.
```

Comment:
Again, it does not mean that this "one process" is correct.
Discussion today :)

Reply:
Fixed